### PR TITLE
IGNITE-22862 Add address information to ClientConnectionException stacktrace

### PR DESCRIPTION
**Pull request by maksaska**
Added some information about client connection to `ClientConnectionException`. With these changes it's easier to analyze connection problems, as it shows the connection description

---------------------------------------------------------------------------------------

Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpClientChannel.java [hunk 1]
```diff
@@ -807,7 +809,7 @@ else if (!supportedVers.contains(srvVer) ||
                 if (e instanceof IOException)
                     err = handleIOError((IOException)e);
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpClientChannel.java:810]: Perhaps `handleIOError` should be fixed too. It uses `sock.toString()`, but sock it's a `ClientConnection` and doesn't implement `toString`.

File: modules/clients/src/test/java/org/apache/ignite/common/NodeSslConnectionMetricTest.java [hunk 2]
```diff
@@ -295,26 +296,32 @@ public void testClientConnector() throws Exception {
         checkSslCommunicationMetrics(reg, 1, 0, 0);
 
         // Tests untrusted certificate.
-        assertThrowsWithCause(() ->
-            startClient(clientConfiguration("client", "trustboth", CIPHER_SUITE, "TLSv1.2")),
+        Throwable ex = assertThrowsWithCause(() ->
+                startClient(clientConfiguration("client", "trustboth", CIPHER_SUITE, "TLSv1.2")),
             ClientConnectionException.class);
 
+        assertContains(log, ex.getMessage(), "127.0.0.1:10800");
```
[the remark this answers is not in the document]
Comment (maksaska) [reply modules/clients/src/test/java/org/apache/ignite/common/NodeSslConnectionMetricTest.java]: I introduced static variable for this test

File: modules/core/src/main/java/org/apache/ignite/internal/client/thin/io/gridnioserver/GridNioClientConnectionMultiplexer.java [hunk 3]
```diff
@@ -194,7 +195,7 @@ public GridNioClientConnectionMultiplexer(ClientConfiguration cfg) {
             return new GridNioClientConnection(ses, msgHnd, stateHnd);
         }
         catch (Exception e) {
-            throw new ClientConnectionException(e.getMessage(), e);
+            throw new ClientConnectionException(e.getMessage() + " [" + S.toString(InetSocketAddress.class, addr) + ']', e);
```
[the remark this answers is not in the document]
Comment (maksaska) [reply modules/core/src/main/java/org/apache/ignite/internal/client/thin/io/gridnioserver/GridNioClientConnectionMultiplexer.java]: I've corrected this message

File: modules/core/src/test/java/org/apache/ignite/client/ReliabilityTest.java [hunk 4]
```diff
@@ -205,7 +205,9 @@ public void testSingleServerFailover() throws Exception {
             // Fail.
             dropAllThinClientConnections(Ignition.allGrids().get(0));
 
-            GridTestUtils.assertThrowsWithCause(() -> cachePut(cache, 0, 0), ClientConnectionException.class);
+            Throwable ex = GridTestUtils.assertThrowsWithCause(() -> cachePut(cache, 0, 0), ClientConnectionException.class);
+
+            GridTestUtils.assertContains(null, ex.getMessage(), cluster.clientAddresses().iterator().next());
```
[the remark this answers is not in the document]
Comment (maksaska) [reply modules/core/src/test/java/org/apache/ignite/client/ReliabilityTest.java]: Done!

File: modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpClientChannel.java (hunk 1 above)
Comment (maksaska) [reply modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpClientChannel.java:811]: Yeah.. I fixed this message

File: modules/core/src/main/java/org/apache/ignite/internal/client/thin/io/gridnioserver/GridNioClientConnectionMultiplexer.java [hunk 5]
```diff
@@ -43,6 +43,7 @@
 import org.apache.ignite.internal.util.nio.GridNioServer;
 import org.apache.ignite.internal.util.nio.GridNioSession;
 import org.apache.ignite.internal.util.nio.ssl.GridNioSslFilter;
+import org.apache.ignite.internal.util.typedef.internal.S;
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/internal/client/thin/io/gridnioserver/GridNioClientConnectionMultiplexer.java:46]: Redundant import

File: modules/core/src/main/java/org/apache/ignite/internal/client/thin/io/gridnioserver/GridNioClientConnectionMultiplexer.java [hunk 6]
```diff
@@ -194,7 +195,7 @@ public GridNioClientConnectionMultiplexer(ClientConfiguration cfg) {
             return new GridNioClientConnection(ses, msgHnd, stateHnd);
         }
         catch (Exception e) {
-            throw new ClientConnectionException(e.getMessage(), e);
+            throw new ClientConnectionException(e.getMessage() + " [" + addr + ']', e);
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/internal/client/thin/io/gridnioserver/GridNioClientConnectionMultiplexer.java:198]: `" [remoteAddress=" + addr`?

File: modules/core/src/test/java/org/apache/ignite/internal/client/thin/FunctionalTest.java [hunk 7]
```diff
@@ -644,6 +645,8 @@ public void testClientFailsOnStart() {
             String.format("%s expected but no exception was received", ClientConnectionException.class.getName()),
             expEx
         );
+
+        assertContains(log, expEx.getMessage(), "[InetSocketAddress [holder=" + Config.SERVER + "]]");
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/internal/client/thin/FunctionalTest.java:649]: This message was changed

File: modules/core/src/main/java/org/apache/ignite/internal/client/thin/io/gridnioserver/GridNioClientConnectionMultiplexer.java (hunk 5 above)
Comment (maksaska) [reply modules/core/src/main/java/org/apache/ignite/internal/client/thin/io/gridnioserver/GridNioClientConnectionMultiplexer.java]: fixed

File: modules/core/src/main/java/org/apache/ignite/internal/client/thin/io/gridnioserver/GridNioClientConnectionMultiplexer.java (hunk 6 above)
Comment (maksaska) [reply modules/core/src/main/java/org/apache/ignite/internal/client/thin/io/gridnioserver/GridNioClientConnectionMultiplexer.java]: Added field to the message

File: modules/core/src/test/java/org/apache/ignite/internal/client/thin/FunctionalTest.java (hunk 7 above)
Comment (maksaska) [reply modules/core/src/test/java/org/apache/ignite/internal/client/thin/FunctionalTest.java]: Fixed test

**Pull request closed without merging**